radicle-httpd/src/api/v1/projects.rs

```diff
@@ -124,3 +214,56 @@ async fn activity_handler(
 
     Ok::<_, Error>((StatusCode::OK, Json(json!({ "activity": timestamps }))))
 }
+
+/// Get project source tree.
+/// `GET /projects/:project/tree/:sha/*path`
+async fn tree_handler(
+    Extension(ctx): Extension<Context>,
+    Path((project, sha, path)): Path<(Id, Oid, String)>,
+) -> impl IntoResponse {
+    let path = path.strip_prefix('/').ok_or(Error::NotFound)?.to_string();
+    let storage = &ctx.profile.storage;
+    let repo = storage.repository(project)?;
+    let tree = radicle_surf::object::tree(
+        &repo.raw().into(),
+        Some(radicle_surf::Revision::Sha { sha }),
+        Some(path),
+    )?;
+    let response = json!({
+        "path": &tree.path,
+        "entries": &tree.entries,
+        "info": &tree.info,
+        "stats": stats(&repo)?,
+    });
+
+    Ok::<_, Error>(Json(response))
+}
+
+#[derive(Serialize)]
+struct Stats {
+    branches: usize,
+    commits: usize,
+    contributors: usize,
+}
+
+fn stats<R: WriteRepository>(repo: &R) -> Result<Stats, Error> {
+    let branches = repo.raw().branches(Some(BranchType::Local))?.count();
+    let (_, head) = repo.head()?;
+    let mut commits = 0;
+    let contributors = History::new(repo.raw().into(), head)?
+        .filter_map(|commit| {
+            commits += 1;
+            if let Ok(commit) = commit {
+                Some((commit.author.name, commit.author.email))
+            } else {
+                None
+            }
+        })
+        .collect::<BTreeSet<_>>();
+
+    Ok(Stats {
+        branches,
```

Comment:
@FintanH I borrowed this code from somewhere in surf I think, why did you not use `HashMap` instead of `BTreeSet`. Wouldn't it be faster?

Reply:
Do you have any evidence of one being faster than the other? :thinking: